### PR TITLE
JENA-2261: Add unit tests for Fuseki UI

### PR DESCRIPTION
**Pull request by kinow**
- [x] fixes annoying message to run npx to update can-i-use dependency
- [x] adds unit tests coverage
- [x] adds unit tests

To run the tests: `yarn run test:unit`, or `mvn clean test`.

To get the coverage reports: `yarn run coverage:unit` (no maven equivalent) then inspect console output, raw `nyc` files in  `.nyc_output`, or HTML reports in `./coverage/index.html`.

What's included (listing what I think more important to get started):

- [x] `ServerStatus.vue`
- [x] `TableListing.vue`
- [x] mixin `table`
- [x] `FusekiService` (interface between the Vue code and Fuseki via Axios library / HTTP)

Views will be tested with the e2e (functional) tests.

Coverage after this PR:

```
=============================== Coverage summary ===============================
Statements   : 78.57% ( 66/84 )
Branches     : 87.5% ( 14/16 )
Functions    : 62.06% ( 18/29 )
Lines        : 78.57% ( 66/84 )
================================================================================
```

The report doesn't include the code in the Views, which is easier to be covered by functional tests. Quite happy with the results so far, so ready for review :tada: :+1: 

image

### CODE REVIEW

File: jena-fuseki2/jena-fuseki-ui/src/components/ServerStatus.vue [hunk 1]
```diff
@@ -61,6 +61,7 @@ export default {
 
   async created () {
     this.serverStatus = await this.$fusekiService.getServerStatus()
+    console.log(this.serverStatus.message)
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-ui/src/components/ServerStatus.vue:64]: Does this print to the console in normal operation?

Comment (kinow) [reply jena-fuseki2/jena-fuseki-ui/src/components/ServerStatus.vue:64]: In the tests output. But if used in a component, then it prints to the browser console (the linter should fail if a console log is added to the main code, unless suppressed).

File: jena-fuseki2/jena-fuseki-ui/vue.config.js [hunk 2]
```diff
@@ -23,6 +24,34 @@ module.exports = {
...
+          .loader('istanbul-instrumenter-loader')
+          .options({ esModules: true })
+          .after('cache-loader')
+      }
+
+      // resolve modules in devtool
+      config.output
+        .devtoolModuleFilenameTemplate('[absolute-resource-path]')
+        .devtoolFallbackModuleFilenameTemplate('[absolute-resource-path]?[hash]')
+    }
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-ui/vue.config.js:54]: This is a mix from some configs from docs about [setting up coverage](https://gist.github.com/lsapan/3bfd0ffc0fb3d4a036fce84f6eea142e) with some copy-paste from a previous project I worked on.

File: jena-fuseki2/jena-fuseki-ui/tests/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  env: {
+    mocha: true,
+  },
+}
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-ui/tests/.eslintrc.js:5]: Mocha is the test framework. Chai is the assertions library... a pair similar to JUnit Hamcrest I think.

File: jena-fuseki2/jena-fuseki-ui/package.json [hunk 4]
```diff
@@ -28,23 +28,30 @@
...
     "eslint-plugin-import": "^2.25.3",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-promise": "^6.0.0",
     "eslint-plugin-standard": "^4.1.0",
     "eslint-plugin-vue": "^8.2.0",
+    "istanbul-instrumenter-loader": "^3.0.1",
+    "nyc": "^15.1.0",
     "sass": "^1.45.2",
     "sass-loader": "^10.1.1",
+    "sinon": "^12.0.1",
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-ui/package.json:53]: Similar to mockito :point_up:

File: jena-fuseki2/jena-fuseki-ui/package.json [hunk 5]
```diff
@@ -28,23 +28,30 @@
...
+    "babel-plugin-istanbul": "^6.1.1",
     "chai": "^4.3.4",
     "eslint": "^7.22.0",
     "eslint-plugin-import": "^2.25.3",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-promise": "^6.0.0",
     "eslint-plugin-standard": "^4.1.0",
     "eslint-plugin-vue": "^8.2.0",
+    "istanbul-instrumenter-loader": "^3.0.1",
+    "nyc": "^15.1.0",
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-ui/package.json:50]: The pair istanbul (instrumenter) and nyc (coverage reporter) are similar to Jacoco or Cobertura.

**Comment by kinow** [comment]
Components:
- [x] `ServerStatus.vue`
- [ ] `dataset/Menu.vue`
- [ ] `manage/Menu.vue`
- [ ] `Menu.vue`
- [ ] `TableListing.vue`

Views will be tested with the e2e (functional) tests.

**Comment by kinow** [comment]
@afs I'll keep adding unit tests whenever I have some spare time. But if you'd like we can merge it at any time (for example, if I disappear due to holidays/work/etc) if you'd like to include this code in a release :+1: 

Only thing worth here, besides tests, is the removal of the Vuex library. I had added it during the initial prototypes as I thought I'd use it (it's like a central storage used by the application, with some concept of transactions/commits for the components/views to share data).

It's not used, so only difference after this change is the size of the artifact that should be a few bytes or kbs smaller.

Cheers
-Bruno

**Review by afs — APPROVED** [review]
Pulled the branch locally. Build works, cleaner than before due to upgrades (:+1:), and yarn.lock is not rewritten.

**Comment by afs** [comment]
> if you'd like to include this code in a release

Yes! In 4.4.0.

**Pull request merged**